kbfsgit/runner_test.go

```diff
@@ -1004,3 +1016,57 @@ func TestRunnerWithKBFSReset(t *testing.T) {
 		rootNode.GetFolderBranch().Tlf, nil, keybase1.MDPriorityGit)
 	require.NoError(t, err)
 }
+
+func testHandlePushBatch(t *testing.T, ctx context.Context,
+	config libkbfs.Config, git, refspec, tlfName string) commitsByRefName {
+	var input bytes.Buffer
+	var output bytes.Buffer
+	r, err := newRunner(ctx, config, "origin",
+		fmt.Sprintf("keybase://private/%s/test", tlfName),
+		filepath.Join(git, ".git"), &input, &output, testErrput{t})
+	require.NoError(t, err)
+
+	args := [][]string{{refspec}}
+	commits, err := r.handlePushBatch(ctx, args)
+	require.NoError(t, err)
+	return commits
+}
+
+func TestRunnerHandlePushBatch(t *testing.T) {
+	ctx, config, tempdir := initConfigForRunner(t)
+	defer libkbfs.CheckConfigAndShutdown(ctx, t, config)
+	defer os.RemoveAll(tempdir)
+
+	git, err := ioutil.TempDir(os.TempDir(), "kbfsgittest")
+	require.NoError(t, err)
+	defer os.RemoveAll(git)
+
+	// Setup the repo
+	h, err := libkbfs.ParseTlfHandle(
+		ctx, config.KBPKI(), config.MDOps(), "user1", tlf.Private)
+	require.NoError(t, err)
+	_, err = libgit.CreateRepoAndID(ctx, config, h, "test")
+	require.NoError(t, err)
+
+	t.Log("Make a new local repo with one commit, and push it. " +
+		"We expect this to return no commits, since it should push the " +
```

Comment:
Was this explicitly decided on?  Seems like it should contain the most recent N commits being pushed, in my opinion.  Might be worth adding this to `pushAll` as well.

Reply:
No, this is until I get a response on how we want `pushAll` to behave.

Reply:
Note that `pushSome` can still be what's used for the initial repo creation (if not all of the local branches are being pushed), so the remote repo could be empty in either case, for what it's worth.

Reply:
That's fine, the code should work either way.